.github/workflows/validate.yml

```diff
@@ -136,8 +136,9 @@ jobs:
              # see https://github.com/actions/virtual-environments/issues/2619#issuecomment-788397841
              sudo /usr/sbin/purge
           fi
-          tar -cvf cabal-head.tar -C $(dirname "$CABAL_EXEC") $(basename "$CABAL_EXEC")
-          echo "CABAL_EXEC_TAR=cabal-head.tar" >> $GITHUB_ENV
+          export CABAL_EXEC_TAR="cabal-head-${{ runner.os }}-x86_64.tar"
```

Comment:
Should it be `.tar` or `.tar.gz` (or `.tar.whatever-else`)?

Reply:
That's a good point. I just checked that `.gz` shrinks the size from 53M to 10M. There's a worry though that Windows users may not have a great support for `.gz`. (I'm not sure about `.tar` either). People used to create `zip`s just based on this in my experience. And I don't know anything about Mac. So, I just left whatever was there before (which was a tar)... Maybe if we have tar, gz shouldn't be a big additional hurdle, and I should do it, thanks!

Reply:
It looks like `tar`ring was added by @jneira who is also our main Windows expert. Javier, do you have any thoughts on `gz`ipping the binary on top of `tar`?

Reply:
I thought the file was already gzipped, just with the wrong extension (below we use `tar -xf …`).

Reply:
`tar -xf` means "eXtract a tar File". In contrast, `tar -xzf` would mean "eXtract a gZipped tar file`.

Fun fact: last week I learned that you can omit `-` in `tar -xf` and alike. How cool is that?!...

Reply:
iirc there was not specific reason to not use gzip and windows support is similar between gz.tar and tar (nowadays people uses 7z which supports both)

Reply:
thanks! will update the pr